### PR TITLE
Fix bug for TAM dates list for new event | Closes #2712

### LINKED ISSUE #2712: Tickets Do Not Appear in Ticket Assignments Manager with New Event on Fresh Install
Here's the data from my `wp_esp_datetime` table

DTT_ID | EVT_ID | name | description | EVT_start | EVT_end | reg_limit | sold | reserved | is_primary | order | parent | deleted
--- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- 
1 | 9 | WSL 1 | yada yada yada | 2020-05-20 12:00:00 | 2020-05-20 21:00:00 | -1 | 0 | 0 | 0 | 1 | 0 | 0

Here's the data from my `wp_esp_ticket` table

TKT_ID | TTM_ID | name | description | qty | sold | reserved | uses | required | min | max | price | start_date | end_date | taxable | order | row | is_default | reverse_calculate | wp_user | parent | deleted
--- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | ---  | --- 
1 | 0 | Free Ticket |  | 100 | 0 | 0 | 	- 1 | 0 | 0 | 	- 1 | 0.000 | 2020 - 04 - 21 00: 12: 07 | 2020 - 05 - 21 03: 59: 00 | 0 | 0 | 1 | 1 | 0 | 1 | 0 | 0

Here's the data from my `wp_esp_datetime_ticket` table

DTK_ID | DTT_ID | TKT_ID
--- | --- | --- 
1 | 1 | 1

And here's a screenie showing the issue in action:

So basically, regardless of which TAM button the user chooses (date relations, ticket relations, or ALL relations) the view should be the same for an event with only a single date and a single ticket.

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR fixes the issue explained [here](https://github.com/eventespresso/event-espresso-core/issues/2712#issuecomment-617104046).

### CODE REVIEW

File: assets/src/application/ui/forms/espressoForm/utils.ts [hunk 1]
```diff
@@ -126,7 +126,7 @@ export const yupToFinalFormErrors = async <T>(validationSchema: ObjectSchema, va
 	} catch (err) {
 		// type annotate the error
 		const error: ValidationError = err;
-		return error.inner.reduce((formError, innerError) => {
+		return error.inner?.reduce((formError, innerError) => {
```
Comment (nicolad) [inline assets/src/application/ui/forms/espressoForm/utils.ts:129]: noice :+1:

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/filters/controls/DatesByMonthControl.tsx [hunk 2]
```diff
@@ -1,19 +1,42 @@
...
+			// if we are lucky to have a next/upcoming month
+			if (nextYearMonth) {
+				setDatesByMonth(nextYearMonth);
+			} else {
+				// as a fallback, set the last one (latest) in the list
+				setDatesByMonth(last<string>(yearMonthOptions));
+			}
+		}
+	}, []);
+
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/ticketAssignmentsManager/filters/controls/DatesByMonthControl.tsx:39]: looks good;
might this snippet be used anywhere else?
was thinking of a custom hook based on this logic

Comment (manzoorwanijk) [reply assets/src/domain/eventEditor/ui/ticketAssignmentsManager/filters/controls/DatesByMonthControl.tsx:39]: That's nowhere else useful IMHO

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils.ts [hunk 3]
```diff
@@ -70,19 +71,39 @@ export const getMonthsListFromDatetimes = (dates: Array<Datetime>): OptionsType
...
 	return list;
 };
+
+export const getYearMonthForNextDate = (dates: Array<Datetime>): string => {
+	const yearWiseMonths = getYearWiseMonthsFromDates(activeUpcoming(dates));
+
+	if (!isEmpty(yearWiseMonths)) {
+		const firstYear = Number(Object.keys(yearWiseMonths)[0]);
+		const firstYearMonths = yearWiseMonths[firstYear];
+		const firstMonth = Number(Object.keys(firstYearMonths)[0]);
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils.ts:103]: or:
```
const [firstYear] = Number(Object.keys(yearWiseMonths));
const [firstMonth] = Number(Object.keys(firstYearMonths));
```
or even better/safer with ramda's `head`:
```
const firstYear = Number(R.head(Object.keys(yearWiseMonths)));
const firstMonth = Number(R.head(Object.keys(firstYearMonths)));
```

Comment (manzoorwanijk) [reply assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils.ts:103]: using `Number` on array will render it useless

Comment (nicolad) [reply assets/src/domain/eventEditor/ui/ticketAssignmentsManager/utils.ts:103]: `head` is returning only first element:
```
R.head(['fi', 'fo', 'fum']); //=> 'fi'
R.head([]); //=> undefined
```

File: assets/src/domain/eventEditor/ui/ticketAssignmentsManager/filters/controls/DatesByMonthControl.tsx (hunk 2 above)
Comment (tn3rb) [reply assets/src/domain/eventEditor/ui/ticketAssignmentsManager/filters/controls/DatesByMonthControl.tsx:39]: > That's nowhere else useful IMHO

he he... hate to inform you but there will be other use cases requiring logic like this 😆 

but that's ok, we just need to remember that this already exists and to extract this when needed.

great job figuring this out though!

**Pull request merged**